### DISCUSSION TITLE
Is Check Mode Simulation?

### DISCUSSION THREAD

**Discussion by MadTooler** [winder/Universal-G-Code-Sender#2071] [Help] [answered]
I think I understand the Check Mode (via toolbox panel) functionality is to make sure the gcode does not have any bad lines and as a visual simulator.  If I am correct that it also functions as a simulator that displays in the visualizer, is there a method to slow it down and speed it up?

**Comment by breiler** [comment, marked as answer]
Yes, the "Check Mode" in GRBL is used to verify the GCode on the actual controller to make sure it will understand the whole program. 
While sending the gcode UGS will display the reported position and the progress of the file being sent.

No, there is no way to slow down or speed up the sent program.

**Reply by MadTooler** [reply]
Thank you.  Understood.  Is there any other method, besides check mode, for a simulation with speed control and gcode line refs within UGS?

**Reply by breiler** [reply]
There is no other way to simulate a file being run in UGS. You can however use the GCode editor to select a line to highlight it in the visualizer. By stepping through line by line you can get a feel for how the gcode progresses.
